tests/unit/jwt-rsa.test.ts

```diff
@@ -114,15 +114,15 @@ describe("unit tests jwt verifier", () => {
         const issuer = "https://example.com";
         const audience = "1234";
         const signedJwt = signJwt(
-          { kid: keypair.jwk.kid, alg: "RS512" },
+          { kid: keypair.jwk.kid, alg: "RS384" },
           { aud: audience, iss: issuer, hello: "world" },
           keypair.privateKey
         );
         const statement = () =>
           verifyJwtSync(signedJwt, keypair.jwk, { issuer, audience });
         expect(statement).toThrow(JwtInvalidSignatureAlgorithmError);
         expect(statement).toThrow(
-          "JWT signature algorithm not allowed: RS512. Expected: RS256"
+          "JWT signature algorithm not allowed: RS384. Expected: RS256"
```

Comment:
I see the unit test doesn't fail, I don't understand why not ... because  I would expect the message to be:

> `"JWT signature algorithm not allowed: RS384. Expected one of: RS256, RS384"`

Do you understand that?

Reply:
I now understand it! This one is thrown because our JWK is set to RS256 (not an array).

Therefore we don't hit the error I was expecting, i.e. in the code that checks the string array matches string (line 193, you changed that to add RS512).

To cover that, I suggest to also have a unit test with a RS384 JWK.

(BTW It is funny we hit 100% code coverage without that, which shows automatic code coverage calculation is not holy, human judgement for the win!)

Or of course, we should support RS384?

Either way I think it would be good to also have a unit test that shows the verification actually works properly with an RS512 signed JWT (and associated JWK).

Reply:
Added a new unit test to validate RS512 verification as part of the unit tests now. This includes generating a JWKS that uses RS512 and validating that the verify operation works. Additionally, as part of this change, I now removed the unit test that was checking for a missing algorithm field in the JWKS, as this would no longer be relevant due to enum definition